commands/publish/index.js

```diff
@@ -298,10 +311,18 @@ class PublishCommand extends Command {
   }
 
   reduceVersions(getVersion) {
-    const iterator = (versionMap, { pkg }) =>
-      Promise.resolve(getVersion(pkg)).then(version => versionMap.set(pkg.name, version));
+    const iterator = ({ versions, hasBreakingChange }, { pkg }) =>
+      Promise.resolve(getVersion(pkg)).then(({ currentVersion, nextVersion }) => {
+        versions.set(pkg.name, nextVersion);
+        /* eslint-disable  no-param-reassign */
+        hasBreakingChange = ConventionalCommitUtilities.isBreakingChange(currentVersion, nextVersion)
+          ? true
+          : hasBreakingChange;
+        /* eslint-enable */
+        return { versions, hasBreakingChange };
```

Comment:
The breaking change detection should be a post-hoc loop, there's no reason to pass along `currentVersion` everywhere, it's still available on each `pkg.version`.